<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SqlExecutionArea</class>
 <widget class="QWidget" name="SqlExecutionArea">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>579</width>
    <height>482</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <property name="spacing">
    <number>3</number>
   </property>
   <property name="leftMargin">
    <number>3</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="rightMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>3</number>
   </property>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="widget">
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="SqlTextEdit" name="editEditor"/>
       </item>
       <item>
        <widget class="QFrame" name="findFrame">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>31</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout">
          <property name="leftMargin">
           <number>1</number>
          </property>
          <property name="topMargin">
           <number>1</number>
          </property>
          <property name="rightMargin">
           <number>1</number>
          </property>
          <property name="bottomMargin">
           <number>1</number>
          </property>
          <property name="spacing">
           <number>3</number>
          </property>
          <item row="0" column="1">
           <widget class="QToolButton" name="previousToolButton">
            <property name="toolTip">
             <string>Find previous match [Shift+F3]</string>
            </property>
            <property name="whatsThis">
             <string>Find previous match with wrapping</string>
            </property>
            <property name="icon">
             <iconset resource="icons/icons.qrc">
              <normaloff>:/icons/up</normaloff>:/icons/up</iconset>
            </property>
            <property name="shortcut">
             <string>Shift+F3</string>
            </property>
           </widget>
          </item>
          <item row="0" column="8">
           <spacer>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </spacer>
          </item>
          <item row="0" column="4">
           <widget class="QCheckBox" name="wholeWordsCheckBox">
            <property name="whatsThis">
             <string>The found pattern must be a whole word</string>
            </property>
            <property name="text">
             <string>Whole Words</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLineEdit" name="findLineEdit">
            <property name="contextMenuPolicy">
             <enum>Qt::DefaultContextMenu</enum>
            </property>
            <property name="whatsThis">
             <string>Text pattern to find considering the checks in this frame</string>
            </property>
            <property name="placeholderText">
             <string>Find in editor</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QCheckBox" name="caseCheckBox">
            <property name="whatsThis">
             <string>The found pattern must match in letter case</string>
            </property>
            <property name="text">
             <string>Case Sensitive</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="nextToolButton">
            <property name="toolTip">
             <string>Find next match [Enter, F3]</string>
            </property>
            <property name="whatsThis">
             <string>Find next match with wrapping</string>
            </property>
            <property name="icon">
             <iconset resource="icons/icons.qrc">
              <normaloff>:/icons/down</normaloff>:/icons/down</iconset>
            </property>
            <property name="shortcut">
             <string>F3</string>
            </property>
           </widget>
          </item>
          <item row="0" column="5">
           <widget class="QCheckBox" name="regexpCheckBox">
            <property name="toolTip">
             <string>Interpret search pattern as a regular expression</string>
            </property>
            <property name="whatsThis">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;When checked, the pattern to find is interpreted as a UNIX regular expression. See &lt;a href=&quot;https://en.wikibooks.org/wiki/Regular_Expressions&quot;&gt;Regular Expression in Wikibooks&lt;/a&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Regular Expression</string>
            </property>
           </widget>
          </item>
          <item row="0" column="6">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="9">
           <widget class="QToolButton" name="hideFindButton">
            <property name="toolTip">
             <string>Close Find Bar</string>
            </property>
            <property name="text">
             <string>Close Find Bar</string>
            </property>
            <property name="icon">
             <iconset resource="icons/icons.qrc">
              <normaloff>:/icons/close</normaloff>:/icons/close</iconset>
            </property>
            <property name="autoRaise">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="ExtendedTableWidget" name="tableResult">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
      </widget>
      <widget class="QTextEdit" name="editErrors">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>120</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <family>Monospace</family>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="acceptDrops">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Results of the last executed statements.&lt;/p&gt;&lt;p&gt;You may want to collapse this panel and use the &lt;span style=&quot; font-style:italic;&quot;&gt;SQL Log&lt;/span&gt; dock with &lt;span style=&quot; font-style:italic;&quot;&gt;User&lt;/span&gt; selection instead.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="whatsThis">
        <string>This field shows the results and status codes of the last executed statements.</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="tabChangesFocus">
        <bool>true</bool>
       </property>
       <property name="undoRedoEnabled">
        <bool>false</bool>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="placeholderText">
        <string>Results of the last executed statements</string>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SqlTextEdit</class>
   <extends>QTextEdit</extends>
   <header>sqltextedit.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ExtendedTableWidget</class>
   <extends>QTableWidget</extends>
   <header>ExtendedTableWidget.h</header>
   <slots>
    <signal>foreignKeyClicked(QString,QString,QByteArray)</signal>
    <signal>foreignKeyClicked(sqlb::ObjectIdentifier,QString,QByteArray)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>editEditor</tabstop>
  <tabstop>findLineEdit</tabstop>
  <tabstop>tableResult</tabstop>
  <tabstop>editErrors</tabstop>
  <tabstop>previousToolButton</tabstop>
  <tabstop>nextToolButton</tabstop>
  <tabstop>caseCheckBox</tabstop>
  <tabstop>wholeWordsCheckBox</tabstop>
 </tabstops>
 <resources>
  <include location="icons/icons.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>saveAsCsv()</slot>
  <slot>saveAsJson()</slot>
  <slot>saveAsView()</slot>
 </slots>
</ui>
